docs/quickstarts/tanstack-start.mdx

```diff
@@ -0,0 +1,243 @@
+---
+title: Tanstack Start Quickstart (Beta)
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+---
+
+> [!WARNING]
+> The TanStack Start SDK is currently in beta. **It is not yet recommended for production use**.
+
+<TutorialHero
+  framework="react"
+  exampleRepo={[
+    {
+      title: "TanStack Start Quickstart Repo",
+      link: "https://github.com/clerk/clerk-tanstack-start-quickstart"
+
+    }
+  ]}
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    }
+  ]}
+>
+  - Install `@clerk/tanstack-start`
+  - Set up your environment variables
+  - Configure `clerkHandler`
+  - Add `<ClerkProvider />` to your application
+  - Protect your pages
+</TutorialHero>
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your TanStack Start application.
+
+<Steps>
+  ### Install `@clerk/tanstack-start`
+
+  Once you have a TanStack Start application ready, you need to install Clerk's TanStack Start SDK. This gives you access to our prebuilt components and hooks for TanStack Start applications.
+
+  <CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ filename: 'terminal' }}
+    npm install @clerk/tanstack-start
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    yarn add @clerk/tanstack-start
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    pnpm add @clerk/tanstack-start
+    ```
+  </CodeBlockTabs>
+
+  ### Set your environment variables
+
+  Add the following keys to your `.env` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+
+  ```env {{ filename: '.env' }}
+  VITE_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  ### Add `clerkHandler` to your application
+
+  To configure Clerk in your TanStack application, you will need to update the SSR entrypoint.
+
+  ```tsx {{ filename: 'app/ssr.tsx', ins: [4, [11, 14]], del: [[6, 9]] }}
+  import { createStartHandler, defaultStreamHandler } from '@tanstack/start/server'
+  import { getRouterManifest } from '@tanstack/start/router-manifest'
+  import { createRouter } from './router'
+  import { createClerkHandler } from '@clerk/tanstack-start/server'
+
+  export default createStartHandler({
+    createRouter,
+    getRouterManifest,
+  })(defaultStreamHandler)
+
+  export default createClerkHandler(
+    createStartHandler({
+      createRouter,
+      getRouterManifest,
+    }),
+  )(defaultStreamHandler)
+  ```
+
+  ### Add `<ClerkProvider>` and components to your app
+
+  All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs/components/clerk-provider) component, which provides active session and user context.
+
+  ```tsx {{ filename: 'app/routes/__root.tsx', ins: [11, 28, 32] }}
```

Comment:
Did you mean to highlight something here? The highlighting is way off.

```suggestion
  ```tsx {{ filename: 'app/routes/__root.tsx', }}
```

Reply:
This probably happened after we enabled prettier formatting for code blocks, I'm gonna fix the highlighting